x-pack/plugin/repository-encrypted/build.gradle

```diff
@@ -8,4 +8,23 @@ esplugin {
     extendedPlugins = ['x-pack-core']
 }
 
+dependencies {
+    // necessary for the license check
+    compileOnly project(path: xpackModule('core'), configuration: 'default')
+    testCompile project(path: xpackModule('core'), configuration: 'testArtifacts')
+    testCompile project(":test:framework")
+    // required for tests of encrypted cloud repositories
+    testCompile project(path: ':plugins:repository-gcs', configuration: 'testArtifacts')
+    testCompile(project(path: ':plugins:repository-azure', configuration: 'testArtifacts'), {
```

Comment:
I don't share your concerns because we're not actually using any of those dependencies in the integ tests in the `repository-encrypted` plugin, as there are no "third party" integ tests. The `repository-encrypted` plugin needs the test classes in the `repository-azure` plugin.

If we were to use those dependencies, we could debate the fragility/messiness of it, but I'm personally relaxed about guava conflicts like these.

I'm going to mark this as "Resolved", feel free to override.